src/main/resources/db/migration/V1.7__Add_Java8features_deck.sql

```diff
@@ -0,0 +1,41 @@
+insert into deck (description, name, rating, category_id, user_id, syntax, created_by) values ('Questions regarding Features that were implemented in Java 8', 'Java 8 Features', 0, 1, 1, 'JAVA', 1);
+
+--SET @deck_id = last_insert_id();
+--SET @deck_id = (SELECT AUTO_INCREMENT FROM  INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'orlp' AND   TABLE_NAME   = 'deck');
+SET @deck_id = (SELECT max(deck_id) from deck where name = 'Java 8 Features');
```

Comment:
This statement selects the last deck_id with name = 'Java 8 features'
The idea was that when our deck would be added to our base, the deck_id would be assigned automatically using auto_increment value therefore we can find it via max() function and name == 'Java 8 Features' is for extra safety because we might not know if there are any other decks that have their deck_id higher than AUTO INCREMENT value or not. 
This can cause problems if on our deploy database we have a deck called 'Java 8 Features' with deck_id higher than auto-increment - then all the cards will be assigned to that deck instead of deck we just created. 
The possible workaround can be found in line 3 where we can use last_insert_id(); to get the value we need but I am not so sure if that is a legit way of doing it so I wanted to ask your opinion as well

Reply:
TL&DR:
1. Yes, you understand it correctly
2. We need the last one because we assume that the deck we just added is the last one in the database
3. We do not know this but I think this should work even without checking the name of the deck, but it adds extra safety